src/main/scala/is/hail/sparkextras/OrderedJoinDistinctIterator.scala

```diff
@@ -0,0 +1,115 @@
+package is.hail.sparkextras
+
+import is.hail.annotations.{JoinedRegionValue, RegionValue}
+
+abstract class OrderedJoinDistinctIterator(leftTyp: OrderedRDD2Type, rightTyp: OrderedRDD2Type, leftIt: Iterator[RegionValue],
+  rightIt: Iterator[RegionValue]) extends Iterator[JoinedRegionValue] {
+  private val jrv = JoinedRegionValue()
+  private var lrv: RegionValue = _
+  private var rrv: RegionValue = if (rightIt.hasNext) rightIt.next() else null
+
+  private var present = false
```

Comment:
I might make this even more explicit it call it `jrvPresent` since there are several things here that could be missing or not.